Sources/Datadog/RUM/RUMMonitor/Scopes/RUMSessionScope.swift

```diff
@@ -14,27 +14,59 @@ internal class RUMSessionScope: RUMScope {
         static let sessionMaxDuration: TimeInterval = 4 * 60 * 60 // 4 hours
     }
 
+    // MARK: - Child Scopes
+
+    /// Active View scopes. Scopes are added / removed when the View starts / stops displaying.
+    private(set) var viewScopes: [RUMScope] = []
```

Comment:
why do we need `[RUMScope]`? i don't see why not `[RUMViewScope]`.?
that would lead to simpler `process(command: OnlyCommandTypeThatConcernsMe)` methods
we won't need to test, for example, `RUMViewScope` with every `RUMCommand` type

Reply:
That's true, we can change it to concrete scope 👌 and this will make few things easier now and later. The idea o `RUMScope` comes from the RFC, but now I don't see any negative impact on testing if we make it a concrete scope (the child scopes creation happens internally, so we don't need / cannot mock the child scopes ✨):

<img width="634" alt="Screenshot 2020-07-24 at 09 36 58" src="https://user-images.githubusercontent.com/2358722/88369881-42081780-cd91-11ea-911c-90d1545ebebf.png">

The unidirectional commands flow can't be enforced more by the architecture, but we keep a good track of it in the implementation and CR 👌.

Reply:
Done in https://github.com/DataDog/dd-sdk-ios/pull/193/commits/42ec45e06c30c2907746ec5873efd65a7ddaec66 for all scopes (Application, Session, View and Resource). This enforces generic `manage(childScopes:...)` anyway, but it reduces the complexity of `as?` casting in some places 👌.